### PR TITLE
update npm 8

### PR DESCRIPTION
**Pull request by Havunen**
- Updated dependencies to latest version
- Updated npm to v8
- fixes broken tests

### CODE REVIEW

File: mbTest/package.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
     "name": "mountebank-tests",
     "version": "0.0.1",
     "author": "Brandon Byars <brandon.byars@gmail.com>",
+    "private": true,
```
Comment (Havunen) [inline mbTest/package.json:5]: added "private" true to avoid releasing this package to npm

File: mbTest/package.json [hunk 2]
```diff
@@ -23,23 +24,22 @@
...
         "nodemailer": "6.7.8",
-        "safe-stable-stringify": "2.3.1",
+        "safe-stable-stringify": "2.4.0",
         "w3cjs": "0.4.0",
         "xpath": "0.0.32"
     },
     "engines": {
-        "node": ">=14",
-        "npm": "~6"
+        "node": ">=14"
```
Comment (Havunen) [inline mbTest/package.json:42]: npm removed as dependency

File: mbTest/web/docsTester/docs.js [hunk 3]
```diff
@@ -81,7 +81,7 @@ function normalizeJSONSubstrings (text) {
 }
 
 function linesOf (text) {
-    return text.replace(/\r/g, '').split('\n');
+    return text.replace(/\r/g, '').split(/\r?\n/);
```
Comment (Havunen) [inline mbTest/web/docsTester/docs.js:84]: support windows and unix style line break style

File: package.json [hunk 4]
```diff
@@ -20,7 +20,6 @@
   "scripts": {
     "jsdoc": "jsdoc src/**/*.js --destination docs --configure .jsdoc --pedantic --readme CONTRIBUTING.md --package package.json",
     "lint": "eslint src/ tasks/ test/ mbTest/ bin/mb && node tasks/lints/licenseCheck.js && node tasks/lints/deadCheck.js && node tasks/lints/objectCheck.js",
-    "postinstall": "cd mbTest && npm install",
```
Comment (Havunen) [inline package.json:23]: removed postinstall to avoid infinite loop between mountebank and mbTest package

File: package.json [hunk 5]
```diff
@@ -63,37 +62,37 @@
...
     "license-checker": "25.0.1",
     "mocha": "10.0.0",
     "mocha-multi-reporters": "1.5.1",
     "nc": "1.0.3",
     "nyc": "^15.1.0",
-    "snyk": "1.986.0",
-    "w3cjs": "0.4.0"
+    "snyk": "1.1012.0",
+    "w3cjs": "0.4.0",
+    "mbTest": "file:./mbTest"
```
Comment (Havunen) [inline package.json:92]: mbTest defined as development dependency in mountebank to have its modules also installed when installing mountebank locally.

File: src/models/tcp/tcpProxy.js [hunk 6]
```diff
@@ -88,7 +88,7 @@ function create (logger, encoding, isEndOfRequest) {
 
                     logger.error(`Proxy ${proxyName} transmission error X=> ${JSON.stringify(error)}`);
 
-                    if (error.code === 'ENOTFOUND') {
+                    if (error.code === 'ENOTFOUND' || error.code === 'EAI_AGAIN') {
```
Comment (Havunen) [inline src/models/tcp/tcpProxy.js:91]: test was failing on windows system

File: src/views/docs/api/overview.ejs [hunk 7]
```diff
@@ -866,6 +866,7 @@ Connection: keep-alive
     "port": <%= port %>,
     "pidfile": "mb.pid",
     "allowInjection": true,
+    "apikey": null,
```
Comment (Havunen) [inline src/views/docs/api/overview.ejs:869]: doc test was failing because apikey was not documented

File: tasks/lints/deadCheck.js [hunk 8]
```diff
@@ -41,7 +41,8 @@ const scan = require('./shared/scan'),
         'mountebank-formatters',
         'nc',
         'nyc',
-        'snyk'
+        'snyk',
+        'mbTest'
```
Comment (Havunen) [inline tasks/lints/deadCheck.js:45]: mbTest added to whitelisted dependencies

**Comment by Havunen** [comment]
@bbyars could you review this PR please

**Comment by bbyars** [comment]
Amazing, thanks for fixing that hack.

**Pull request merged**